app/src/main/java/com/example/task_planner_app/ui/fragments/HomeFragment.kt

```diff
@@ -0,0 +1,117 @@
+package com.example.task_planner_app.ui.fragments
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.fragment.app.viewModels
+import androidx.navigation.fragment.findNavController
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.task_planner_app.R
+import com.example.task_planner_app.databinding.FragmentHomeBinding
+import com.example.task_planner_app.repository.remote.dto.TaskDto
+import com.example.task_planner_app.ui.adapter.TaskAdapter
+import com.example.task_planner_app.ui.listeners.OnDeleteListener
+import com.example.task_planner_app.viewmodel.MainActivityViewModel
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+class HomeFragment : Fragment() {
+
+    private lateinit var binding: FragmentHomeBinding
+    //private lateinit var taskAdapter : TaskAdapter
+    lateinit var taskAdapter: TaskAdapter
+    //private val taskAdapter = TaskAdapter()
+    private val viewModel by viewModels<MainActivityViewModel>()
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        // Inflate the layout for this fragment
+        binding = FragmentHomeBinding.inflate(inflater, container, false)
+
+        // Init view
+        initView()
```

Comment:
I suggest to call this 3 functions on the onViewCreated function which you can override too.